telosnetwork/open-block-explorer
# #542 | Fix send token balance

# Fixes #542

## Description
This PR fixes an issue where the available balance shown is not accurate in the Send dialogue. This is due to two issues, 1. it is possible for the user to open the Send dialogue before the available tokens list has loaded, 2. the token balance being determined by comparing the token symbol alone, despite that there are several tokens with the same symbol and different contracts. The fix was to add a loading state until tokens are loaded, and then determine token balance based on token symbol _and_ contract name.

## Test scenarios
- note: first part of this must be tested locally, as there is no `develop` deploy preview for this PR
- go to http://localhost:8080
- login using cleos: account: **montevideouy**, permission: **active**
- go to http://localhost:8080/account/montevideouy
    - the left button under account balances should say "Loading..." and be disabled on page load, then switch to "Send" and be enabled after a few seconds
- take notice of the liquid balance (~87 TLOS)
- click Send
    - the available balance on the TLOS input should match the liquid balance from previous screen
- go to https://deploy-preview-555--open-block-explorer-dev.netlify.app and login using the credentials above
- go to an account with 0 TLOS, e.g. https://deploy-preview-555--open-block-explorer-dev.netlify.app/account/montevideouy
- click Send
    - the input should have "--" above it as no token is selected
- select a token from the dropdown
    - the balance of that token should show above the input

## Checklist:
